src/models/transfers.js

```diff
@@ -250,28 +270,23 @@ function * fulfillTransfer (transferId, fulfillment) {
     }
 
     const isExecution = validateConditionFulfillment(transfer, fulfillment)
-    const accountBalances = yield makeAccountBalances(transaction, transfer)
+
+    if (isExecution && transfer.state === 'executed' || !isExecution && transfer.state === 'rejected') {
+      return (yield fulfillments.getFulfillment(transferId)).getDataExternal()
```

Comment:
I can't think of a clean way to make this clearer, so I'm inclined to leave as is for now. It's not perfect, but I'd prefer to revisit later (if we decide it needs refactoring) than be held up now.